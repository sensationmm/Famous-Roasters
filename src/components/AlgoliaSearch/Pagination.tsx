import { ArrowNarrowLeftIcon, ArrowNarrowRightIcon } from '@heroicons/react/solid'
import { useEffect } from 'react'
import { usePagination, UsePaginationProps } from 'react-instantsearch-hooks-web'

const getButtonClassNames = (disabled: boolean): string => {
  const classNames: string[] = [
    'border-t-2',
    'border-transparent',
    'pt-4',
    'pr-1',
    'inline-flex',
    'items-center',
    'font-bold',
  ]

  // disabled
  if (disabled) {
    classNames.push('text-coreUI-text-tertiary')
  } else {
    classNames.push('text-coreUI-text-secondary', 'hover:text-coreUI-text-primary')
  }

  return classNames.join(' ')
}

const Pagination = (props: UsePaginationProps) => {
  let scrollTimeout: ReturnType<typeof setTimeout>
  const { currentRefinement, isFirstPage, isLastPage, refine, pages } = usePagination(props)

  useEffect(() => {
    return clearTimeout(scrollTimeout)
  }, [])

  const refineWithScrollToTop = (x: number) => {
    refine(x)

    const mediaQuery = window.matchMedia('(prefers-reduced-motion: reduce)')
    scrollTimeout = setTimeout(
      () =>
        window.scroll({
          top: 0,
          behavior: !mediaQuery || mediaQuery.matches ? 'auto' : 'smooth',
        }),
      100,
    )
  }

  return (
    <nav className="border-t border-coreUI-border flex items-center justify-between mb-8">
      <div className="flex-1 flex justify-end mr-2">
        <button
          onClick={() => refineWithScrollToTop(currentRefinement - 1)}
          disabled={isFirstPage}
          className={getButtonClassNames(isFirstPage)}
          data-testid="pagination-previous"
          name="Previous Page"
        >
          <ArrowNarrowLeftIcon className="mr-1.5 h-5 w-5" aria-hidden="true" />
        </button>
      </div>

      {pages.map((page, count) => (
        <div
          data-testid={`pagination-page-${count}`}
          onClick={() => refineWithScrollToTop(page)}
          key={page}
          className={`cursor-pointer w-12 h-10 pt-3.5 text-center font-bold text-coreUI-text-tertiary border-t-2 ${
            currentRefinement === page ? ' text-coreUI-text-primary border-black' : ' border-transparent'
          }`}
        >
          {page + 1}
        </div>
      ))}

      <div className="flex-1 flex justify-start -mt-0.5 ml-2">
        <button
          onClick={() => refineWithScrollToTop(currentRefinement + 1)}
          disabled={isLastPage}
          className={getButtonClassNames(isLastPage)}
          data-testid="pagination-next"
          name="Next Page"
        >
          <ArrowNarrowRightIcon className="ml-1.5 h-5 w-5" aria-hidden="true" />
        </button>
      </div>
    </nav>
  )
}

export default Pagination
