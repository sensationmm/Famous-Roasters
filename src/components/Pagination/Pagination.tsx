import { ArrowNarrowLeftIcon, ArrowNarrowRightIcon } from '@heroicons/react/solid'
import React from 'react'
import { useTranslation } from 'react-i18next'

interface PaginationProps {
  hasNextPage: boolean
  hasPreviousPage: boolean
  next: () => void
  previous: () => void
}

const getButtonClassNames = (disabled: boolean): string => {
  const classNames: string[] = [
    'border-t-2',
    'border-transparent',
    'pt-4',
    'pr-1',
    'inline-flex',
    'items-center',
    'text-sm',
    'font-bold',
  ]

  // disabled
  if (disabled) {
    classNames.push('text-coreUI-text-tertiary')
  } else {
    classNames.push('text-coreUI-text-secondary', 'hover:text-coreUI-text-primary')
  }

  return classNames.join(' ')
}

export const Pagination: React.FC<PaginationProps> = ({ hasNextPage, hasPreviousPage, next, previous }) => {
  const { t } = useTranslation()

  return (
    <nav className="border-t border-coreUI-border flex items-center justify-between mb-4">
      <div className="flex-1 flex justify-end -mt-0.5 mr-2">
        <button
          onClick={previous}
          disabled={!hasPreviousPage}
          className={getButtonClassNames(!hasPreviousPage)}
          data-testid="pagination-previous"
        >
          <ArrowNarrowLeftIcon className="mr-1.5 h-5 w-5" aria-hidden="true" />
          <span>{t('pages.catalogue.pagination.previous')}</span>
        </button>
      </div>
      <div className="flex-1 flex justify-start -mt-0.5 ml-2">
        <button
          onClick={next}
          disabled={!hasNextPage}
          className={getButtonClassNames(!hasNextPage)}
          data-testid="pagination-next"
        >
          <span>{t('pages.catalogue.pagination.next')}</span>
          <ArrowNarrowRightIcon className="ml-1.5 h-5 w-5" aria-hidden="true" />
        </button>
      </div>
    </nav>
  )
}
