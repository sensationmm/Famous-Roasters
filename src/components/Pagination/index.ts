import { Pagination } from './Pagination'

export { Pagination as Pagination }
