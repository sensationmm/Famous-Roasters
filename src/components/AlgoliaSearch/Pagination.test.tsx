import { render, waitFor } from '@testing-library/react'
import React from 'react'
import { InstantSearch } from 'react-instantsearch-hooks-web'

import Pagination from './Pagination'

describe('Pagination component', () => {
  const snippet = () => (
    <InstantSearch
      indexName="products"
      searchClient={{
        async search() {
          return {
            results: [
              {
                hits: [
                  { id: 'result1' } as any,
                  { id: 'result2' } as any,
                  { id: 'result3' } as any,
                  { id: 'result4' } as any,
                  { id: 'result5' } as any,
                  { id: 'result6' } as any,
                  { id: 'result7' } as any,
                  { id: 'result8' } as any,
                  { id: 'result9' } as any,
                  { id: 'result10' } as any,
                ],
                page: 2,
                nbHits: 10,
                nbPages: 3,
                hitsPerPage: 3,
                processingTimeMS: 1,
                exhaustiveNbHits: true,
                query: '',
                params: '',
              },
            ],
          }
        },
      }}
    >
      <Pagination />
    </InstantSearch>
  )

  it('Renders correctly', async () => {
    const { container } = render(snippet())
    await waitFor(() => new Promise((res) => setTimeout(res, 0)))
    expect(container).toMatchSnapshot()
  })
})
