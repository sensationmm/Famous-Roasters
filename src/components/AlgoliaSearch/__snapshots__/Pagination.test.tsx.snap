// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination component Renders correctly 1`] = `
<div>
  <nav
    class="border-t border-coreUI-border flex items-center justify-between mb-8"
  >
    <div
      class="flex-1 flex justify-end mr-2"
    >
      <button
        class="border-t-2 border-transparent pt-4 pr-1 inline-flex items-center font-bold text-coreUI-text-tertiary"
        data-testid="pagination-previous"
        disabled=""
      >
        <svg
          aria-hidden="true"
          class="mr-1.5 h-5 w-5"
          fill="currentColor"
          viewBox="0 0 20 20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M7.707 14.707a1 1 0 01-1.414 0l-4-4a1 1 0 010-1.414l4-4a1 1 0 011.414 1.414L5.414 9H17a1 1 0 110 2H5.414l2.293 2.293a1 1 0 010 1.414z"
            fill-rule="evenodd"
          />
        </svg>
      </button>
    </div>
    <div
      class="cursor-pointer w-12 h-10 pt-3.5 text-center font-bold text-coreUI-text-tertiary border-t-2  text-coreUI-text-primary border-black"
      data-testid="pagination-page-0"
    >
      1
    </div>
    <div
      class="flex-1 flex justify-start -mt-0.5 ml-2"
    >
      <button
        class="border-t-2 border-transparent pt-4 pr-1 inline-flex items-center font-bold text-coreUI-text-tertiary"
        data-testid="pagination-next"
        disabled=""
      >
        <svg
          aria-hidden="true"
          class="ml-1.5 h-5 w-5"
          fill="currentColor"
          viewBox="0 0 20 20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M12.293 5.293a1 1 0 011.414 0l4 4a1 1 0 010 1.414l-4 4a1 1 0 01-1.414-1.414L14.586 11H3a1 1 0 110-2h11.586l-2.293-2.293a1 1 0 010-1.414z"
            fill-rule="evenodd"
          />
        </svg>
      </button>
    </div>
  </nav>
</div>
`;
